concordium-std/src/impls.rs

```diff
@@ -11,6 +11,13 @@ use crate::{
 use concordium_contracts_common::*;
 use mem::MaybeUninit;
 
+#[cfg(not(feature = "std"))]
+use core::hash;
+#[cfg(feature = "std")]
+use std::hash;
```

Comment:
Distinguishing imports based on features should generally be done in `lib.rs` so as to not pollute the other files. The imports can be re-exported either with pub or pub(crate)